### PR TITLE
CORE-8735: Bypass membership worker for quering the registration status

### PR DESCRIPTION
**Pull request by yift-r3**
By-pass the membership worker to get the registration status. 

API pull request in https://github.com/corda/corda-api/pull/845

### CODE REVIEW

File: components/membership/membership-client-impl/src/main/kotlin/net/corda/membership/impl/client/MemberOpsClientImpl.kt [hunk 1]
```diff
@@ -299,118 +265,45 @@ class MemberOpsClientImpl @Activate constructor(
...
-                logger.warn(
-                    "Could not check statuses of registration requests made by holding identity ID" +
-                            " [${holdingIdentityShortHash}].", e
-                )
-                emptyList()
+            val holdingIdentity = virtualNodeInfoReadService.getByHoldingIdentityShortHash(holdingIdentityShortHash)
+                ?: throw CouldNotFindMemberException(holdingIdentityShortHash)
+            return membershipQueryClient.queryRegistrationRequestsStatus(
+                holdingIdentity.holdingIdentity
+            ).getOrThrow().map {
```
Comment (charlieR3) [inline components/membership/membership-client-impl/src/main/kotlin/net/corda/membership/impl/client/MemberOpsClientImpl.kt]: Is the `QueryException` that this throws handled anywhere or will it just result in a 500 error on the API?

Comment (charlieR3) [reply components/membership/membership-client-impl/src/main/kotlin/net/corda/membership/impl/client/MemberOpsClientImpl.kt]: Same question on the function changed below this

Comment (yift-r3) [reply components/membership/membership-client-impl/src/main/kotlin/net/corda/membership/impl/client/MemberOpsClientImpl.kt]: I think it will throw 500.

**Review by charlieR3 — DISMISSED** [review]

**Review by charlieR3 — DISMISSED** [review]

**Review by charlieR3 — DISMISSED** [review]

**Review by charlieR3 — DISMISSED** [review]

File: components/membership/membership-client-impl/src/main/kotlin/net/corda/membership/impl/client/MemberOpsClientImpl.kt [hunk 2]
```diff
@@ -178,21 +164,8 @@ class MemberOpsClientImpl @Activate constructor(
             }
 
             is ConfigChangedEvent -> {
-                impl.close()
```
Comment (owenstanford) [inline components/membership/membership-client-impl/src/main/kotlin/net/corda/membership/impl/client/MemberOpsClientImpl.kt:181]: The Publisher is a resource, we should call close as we do for RPC

Comment (yift-r3) [reply components/membership/membership-client-impl/src/main/kotlin/net/corda/membership/impl/client/MemberOpsClientImpl.kt:181]: I removed the publisher (the publisher wasn't a resource when it was written)

File: components/membership/membership-http-rpc-impl/src/test/kotlin/net/corda/membership/impl/httprpc/v1/MemberRegistrationRestResourceTest.kt [hunk 3]
```diff
@@ -160,6 +163,34 @@ class MemberRegistrationRestResourceTest {
...
+        memberRegistrationRpcOps.start()
+        memberRegistrationRpcOps.activate("")
+
+        assertThrows<ResourceNotFoundException> {
+            memberRegistrationRpcOps.checkRegistrationProgress(HOLDING_IDENTITY_ID)
+        }
+    }
+
+    @Test
+    fun `checkRegistrationProgress throw 503 when the database is not readt`() {
```
Comment (owenstanford) [inline components/membership/membership-http-rpc-impl/src/test/kotlin/net/corda/membership/impl/httprpc/v1/MemberRegistrationRestResourceTest.kt]: small typo in test name

**Review by owenstanford — COMMENTED** [review]
- For the REST API it looks good.

- Think you need to fix the ActiveImpl so it's disposed and closes out the producer when a ConfigChangedEvent is received.

**Review by charlieR3 — APPROVED** [review]

**Review by owenstanford — APPROVED** [review]

**Pull request merged**